app.rb

```diff
@@ -0,0 +1,115 @@
+require_relative 'rental'
+require_relative 'person'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def start
+    puts 'Welcome to the Library System!'
+    list_of_options
+  end
+
+  def list_all_books
+    if @books.length.positive?
+      @books.each do |book|
+        puts "Title: #{book.title}, Author: #{book.author}"
+      end
+    else
+      puts 'No books found'
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    person_type = gets.chomp
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'That is not a valid input'
+      create_person
+    end
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]'
+    parent_permission = gets.chomp
+    parent_permission = parent_permission.downcase == 'y'
+    student = Student.new(nil, age, name, parent_permission)
+    @people << student
+    puts 'Person(Student) created successfully'
+  end
```

Comment:
Kindly fix the error when creating a student to make sure that persons are created successfully.

![image](https://user-images.githubusercontent.com/38088818/211133990-535b016e-62cc-4314-9b78-f9db2f9b038e.png)

- The issue is with the parameters in the `initialize` method in the `student.rb` class. The expected arguments are 2..3 and 4 were passed. Kindly make sure that you set up your parameters correctly.

Creating a teacher works perfectly though. Great job!!! 👍🏾

Reply:
Fixed.